File: 1180-count-substrings-with-only-one-distinct-letter/1180-count-substrings-with-only-one-distinct-letter.py
class Solution:
    def countLetters(self, s: str) -> int:
        total, curr = 1, 1
        for i in range(1, len(s)):
            if s[i] == s[i - 1]:
                curr += 1
            else:
                curr = 1
            total += curr
        return total

File: 1180-count-substrings-with-only-one-distinct-letter/1180-count-substrings-with-only-one-distinct-letter.cpp
class Solution {
public:
    int countLetters(string s) {
        int total = 1, curr = 1;
        for (int i = 1; i < s.length(); ++i) {
            if (s[i] == s[i - 1]) ++curr;
            else curr = 1;
            total += curr;
        }
        return total;
    }
};